zulipterminal/ui_tools/boxes.py

```diff
@@ -163,6 +163,15 @@ def _time_for_message(self, message: Dict[str, Any]) -> str:
         return ctime(message['timestamp'])[:-8]
 
     def need_recipient_header(self) -> bool:
+        # Prevent redundant information in recipient bar
+        # and recipient header.
+        if len(self.model.narrow) == 1 and \
+                self.model.narrow[0][0] == 'pm_with':
+            return False
+        if len(self.model.narrow) == 2 and \
+                self.model.narrow[1][0] == 'topic':
+            return False
```

Comment:
I see what you've achieved with this commit and I think it's useful.
Does this just pass an existing test? You've not added a test to check this directly?

I wanted to clarify that this wasn't quite what I meant with hiding the recipient when it's "fully narrowed"; in that case we can of course do something like you have done here, but also we can merge the two boxes together (or hide the bottom one, perhaps).

Reply:
This does pass the existing test. However I did not find any tests specifically for this. 
 
Yup. I understood what you mean earlier. We can make this a v2 if you agree, and merge the two boxes together, in specific narrows.

Reply:
@neiljp Do you want me to add tests for this? There are no previous tests for `need_recipient_header` If I am not wrong.

Reply:
Agreed; when merged it will be easier to get feedback :)